gallery/src/styles/NxFormLayout/NxFormInlineLayoutExample.tsx

```diff
@@ -4,43 +4,65 @@
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import React, { FormEvent } from 'react';
+import React, { useState } from 'react';
 import {faPlus, faMinus} from '@fortawesome/free-solid-svg-icons';
 
-import { NxButton, NxFormGroup } from '@sonatype/react-shared-components';
-import { NxStatefulTextInput } from '@sonatype/react-shared-components';
-import { NxFontAwesomeIcon } from '@sonatype/react-shared-components';
+import {
+  NxFontAwesomeIcon,
+  NxButton,
+  NxButtonBar,
+  NxFormGroup,
+  NxFormRow,
+  nxTextInputStateHelpers,
+  NxStatefulForm,
+  NxTextInput,
+  hasValidationErrors,
+  combineValidationErrors
+} from '@sonatype/react-shared-components';
 
 export default function NxFormLayoutExample() {
   function validator(val: string) {
     return val.length ? null : 'Must be non-empty';
   }
 
-  function onSubmit(evt: FormEvent) {
-    evt.preventDefault();
+  const [usernameState, setUsernameState] = useState(nxTextInputStateHelpers.initialState('', validator)),
+      [hostnameState, setHostnameState] = useState(nxTextInputStateHelpers.initialState('', validator)),
+      formValidationErrors = hasValidationErrors(combineValidationErrors(
+          usernameState.validationErrors,
+          hostnameState.validationErrors
+      )) ? 'Required fields are missing' : null;
+
+  function onUsernameChange(val: string) {
+    setUsernameState(nxTextInputStateHelpers.userInput(validator, val));
+  }
+
+  function onHostnameChange(val: string) {
+    setHostnameState(nxTextInputStateHelpers.userInput(validator, val));
+  }
+
+  function onSubmit() {
     alert('Submitted!');
   }
 
   return (
-    <form className="nx-form" onSubmit={onSubmit} aria-label="Inline Form Example">
-      <div className="nx-form-row">
+    <NxStatefulForm onSubmit={onSubmit} validationErrors={formValidationErrors} aria-label="Inline Form Example">
+      <NxFormRow>
         <NxFormGroup label="Username" isRequired>
-          <NxStatefulTextInput aria-required={true} validator={validator}/>
+          <NxTextInput { ...usernameState } validatable onChange={onUsernameChange} />
         </NxFormGroup>
-        <div className="nx-btn-bar">
-          <NxButton variant="primary" type="submit">Submit</NxButton>
-        </div>
-      </div>
-      <div className="nx-form-row">
+        <NxButtonBar>
+          <NxButton variant="tertiary" type="button">Query</NxButton>
+        </NxButtonBar>
+      </NxFormRow>
+      <NxFormRow>
         <NxFormGroup label="Hostname" sublabel="Add a hostname" isRequired>
-          <NxStatefulTextInput aria-required={true}
-                               validator={validator}/>
+          <NxTextInput { ...hostnameState } validatable onChange={onHostnameChange} />
         </NxFormGroup>
-        <div className="nx-btn-bar">
+        <NxButtonBar>
           <NxButton title="Add" variant="icon-only"><NxFontAwesomeIcon icon={faPlus}/></NxButton>
```

Comment:
This is a small thing but I wonder if we should add `type="button"` to these  (plus minus buttons) so they don't submit the form when you click them.